src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -437,26 +432,59 @@ public void setConcurrentBuild(boolean b) throws IOException {
     }
 
     @Override public Map<TriggerDescriptor, Trigger<?>> getTriggers() {
-        return triggers.toMap();
+        return getTriggersJobProperty().getTriggersMap();
     }
 
-    public void addTrigger(Trigger trigger) {
-        Trigger old = triggers.get(trigger.getDescriptor());
-        if (old != null) {
-            old.stop();
-            triggers.remove(old);
+    public PipelineTriggersJobProperty getTriggersJobProperty() {
```

Comment:
Actually, no, it can't - needed in the `TransientActionFactory` for `PipelineTriggersJobProperty`.